lib/omniauth/strategies/jwt.rb

```diff
@@ -25,7 +25,7 @@ def request_phase
       
       def decoded
         begin
-          @decoded ||= ::JWT.decode(request.params['jwt'], options.secret, options.algorithm).first
+          @decoded ||= ::JWT.decode(request.params['jwt'], options.secret, true, options.decode_options).first
```

Comment:
The `.decode` call was always wrong but we were lucky and `HS256` is truthy so we **were** verifying the signature... But the algorithm should be passed as an option and not a parameter.